### ISSUE TITLE
Getting error after add this plugin to project

### ISSUE THREAD

**Issue by alexeynobody** [ably/ably-flutter#22]
Hello, after add this plugin to .pubspec:

```
  ably_flutter_plugin:
    git: https://github.com/ably/ably-flutter.git
```

I am getting errors during build:
```
Launching lib/main.dart on iPhone 11 in debug mode...
Running pod install...                                              5,5s
Running Xcode build...                                                  
                                                   
 └─Compiling, linking and signing...                         1,8s
Xcode build done.                                           32,4s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ios/Pods/AblyDeltaCodec/xdelta/xdelta3/xdelta3.c:354:1: warning: unused function
    'xd3_rlist_length' [-Wunused-function]
    XD3_MAKELIST(xd3_rlist, xd3_rinst, link);
    ^
    In file included from ios/Pods/AblyDeltaCodec/xdelta/xdelta3/xdelta3.c:351:
    ios/Pods/AblyDeltaCodec/xdelta/xdelta3/xdelta3-list.h:114:73: note: expanded from
    macro 'XD3_MAKELIST'
    static inline usize_t                                                   \
                                                                            ^
    <scratch space>:25:1: note: expanded from here
    xd3_rlist_length
    ^
    ios/Pods/AblyDeltaCodec/xdelta/xdelta3/xdelta3.c:479:20: warning: unused function
    'xd3_emit_uint32_t' [-Wunused-function]
    static int         xd3_emit_uint32_t (xd3_stream *stream, xd3_output **output,
                       ^
    2 warnings generated.
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:43:53: warning: implicit boolean conversion of Objective-C object literal always evaluates to true [-Wobjc-literal-conversion]
        WRITE_VALUE(dictionary, TxErrorInfo_statusCode, @([e statusCode]));
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
        ~~  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:54:62: warning: implicit boolean conversion of Objective-C object literal always evaluates to true [-Wobjc-literal-conversion]
        WRITE_VALUE(dictionary, TxConnectionStateChange_retryIn, @((int)([stateChange retryIn] * 1000)));
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
        ~~  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:51:61: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionState' (aka 'enum ARTRealtimeConnectionState')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_current, [stateChange current]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:51:61: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionState' (aka 'enum ARTRealtimeConnectionState')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_current, [stateChange current]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:52:62: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionState' (aka 'enum ARTRealtimeConnectionState')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_previous, [stateChange previous]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:52:62: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionState' (aka 'enum ARTRealtimeConnectionState')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_previous, [stateChange previous]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:53:59: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionEvent' (aka 'enum ARTRealtimeConnectionEvent')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_event, [stateChange event]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:53:59: warning: implicit conversion loses integer precision: 'ARTRealtimeConnectionEvent' (aka 'enum ARTRealtimeConnectionEvent')
    to 'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxConnectionStateChange_event, [stateChange event]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:64:59: warning: implicit boolean conversion of Objective-C object literal always evaluates to true [-Wobjc-literal-conversion]
        WRITE_VALUE(dictionary, TxChannelStateChange_resumed, @([stateChange resumed]));
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
        ~~  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:61:58: warning: implicit conversion loses integer precision: 'ARTRealtimeChannelState' (aka 'enum ARTRealtimeChannelState') to
    'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_current, [stateChange current]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:61:58: warning: implicit conversion loses integer precision: 'ARTRealtimeChannelState' (aka 'enum ARTRealtimeChannelState') to
    'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_current, [stateChange current]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:62:59: warning: implicit conversion loses integer precision: 'ARTRealtimeChannelState' (aka 'enum ARTRealtimeChannelState') to
    'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_previous, [stateChange previous]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:62:59: warning: implicit conversion loses integer precision: 'ARTRealtimeChannelState' (aka 'enum ARTRealtimeChannelState') to
    'int' [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_previous, [stateChange previous]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:63:56: warning: implicit conversion loses integer precision: 'ARTChannelEvent' (aka 'enum ARTChannelEvent') to 'int'
    [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_event, [stateChange event]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:29:9: note: expanded from macro 'WRITE_VALUE'
        if (VALUE) { \
            ^~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:63:56: warning: implicit conversion loses integer precision: 'ARTChannelEvent' (aka 'enum ARTChannelEvent') to 'int'
    [-Wshorten-64-to-32]
        WRITE_ENUM(dictionary, TxChannelStateChange_event, [stateChange event]);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:36:67: note: expanded from macro 'WRITE_ENUM'
            WRITE_VALUE(DICTIONARY, JSON_KEY, [NSNumber numberWithInt:ENUM_VALUE]); \
                                              ~                       ^~~~~~~~~~
    /Users/alexey/Documents/dev/flutter/.pub-cache/git/ably-flutter-0c2f7074b7ffe6abcb5ba0a0a99aed7ab4917bd9/ios/Classes/codec/AblyFlutterWri
    ter.m:30:31: note: expanded from macro 'WRITE_VALUE'
            [DICTIONARY setObject:VALUE forKey:JSON_KEY]; \
                                  ^~~~~
    15 warnings generated.
    3 warnings generated.
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:9: error: include of non-modular header inside framework module 'ably_flutter_plugin.AblyFlutterReader':
    'build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h'
    [-Werror,-Wnon-modular-include-in-framework-module]
    #import "ARTTokenDetails.h"
            ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:18:1:
    error: duplicate interface definition for class 'ARTTokenDetails'
    @interface ARTTokenDetails : NSObject<NSCopying>
    ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:18:12:
    note: previous definition is here
    @interface ARTTokenDetails : NSObject<NSCopying>
               ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:23:49:
    error: property has a previous declaration
    @property (nonatomic, readonly, copy) NSString *token;
                                                    ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:23:49:
    note: property declared here
    @property (nonatomic, readonly, copy) NSString *token;
                                                    ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:28:59:
    error: property has a previous declaration
    @property (nonatomic, readonly, strong, nullable) NSDate *expires;
                                                              ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:28:59:
    note: property declared here
    @property (nonatomic, readonly, strong, nullable) NSDate *expires;
                                                              ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:33:59:
    error: property has a previous declaration
    @property (nonatomic, readonly, strong, nullable) NSDate *issued;
                                                              ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:33:59:
    note: property declared here
    @property (nonatomic, readonly, strong, nullable) NSDate *issued;
                                                              ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:38:59:
    error: property has a previous declaration
    @property (nonatomic, readonly, copy, nullable) NSString *capability;
                                                              ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:38:59:
    note: property declared here
    @property (nonatomic, readonly, copy, nullable) NSString *capability;
                                                              ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:43:59:
    error: property has a previous declaration
    @property (nonatomic, readonly, copy, nullable) NSString *clientId;
                                                              ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:43:59:
    note: property declared here
    @property (nonatomic, readonly, copy, nullable) NSString *clientId;
                                                              ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:53:12:
    error: definition of 'ARTTokenDetails' must be imported from module 'Ably.ARTTokenDetails' before it is required
    @interface ARTTokenDetails (ARTTokenDetailsCompatible) <ARTTokenDetailsCompatible>
               ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:18:12:
    note: previous definition is here
    @interface ARTTokenDetails : NSObject<NSCopying>
               ^
    While building module 'ably_flutter_plugin' imported from
    ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/ably_flutter_plugin-umbrella.h:18:
    In file included from
    build/ios/Debug-iphonesimulator/ably_flutter_plugin/ably_flutter_plugin.framework/Head
    ers/AblyFlutterReader.h:3:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:53:29:
    warning: duplicate definition of category 'ARTTokenDetailsCompatible' on interface 'ARTTokenDetails'
    @interface ARTTokenDetails (ARTTokenDetailsCompatible) <ARTTokenDetailsCompatible>
                                ^
    In module 'Ably' imported from
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:10:
    build/ios/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTTokenDetails.h:53:12:
    note: previous definition is here
    @interface ARTTokenDetails (ARTTokenDetailsCompatible) <ARTTokenDetailsCompatible>
               ^
    1 warning and 8 errors generated.
    ios/Runner/GeneratedPluginRegistrant.m:8:9: fatal error: could not build module
    'ably_flutter_plugin'
    #import <ably_flutter_plugin/AblyFlutterPlugin.h>
     ~~~~~~~^
    1 warning and 9 errors generated.
```

What am I doing wrong?

**Comment by mattheworiordan** [comment]
Hi @alexeynobody.  Thanks for reporting this issue.
We are looking into this and will aim to get a response back to you as quickly as possible (in the next few days at most).

**Comment by alexeynobody** [comment]
Any news?

**Comment by tiholic** [comment]
@alexeynobody gladly, we were able to reproduce the issue on **MacOS Catalina(10.15.5 (19F101))** and fixed it by upgrading to latest version of **XCode (Version 11.6 (11E708))**

You may explore more on this here:
1. https://github.com/flutter/flutter/issues/43713
2. https://github.com/flutter/flutter/issues/59919

**Comment by alexeynobody** [comment]
I use latest stable Xcode on MacOS Catalina 10.15.6 (19G73) and this error still here

**Comment by alexeynobody** [comment]
Example app work correctly, maybe example app use specific settings for build?

**Comment by alexeynobody** [comment]
@tiholic after remove this lines from Podfile:
```
  use_frameworks!
  use_modular_headers!
```

All works great

A found it in example. Why does this fix the build after removing a few commands?

**Comment by alexeynobody** [comment]
Any news?

**Comment by tiholic** [comment]
@alexeynobody this issue was addressed in #46
Feel free to re-open the issue if its still reproducible

cc: @QuintinWillison
